### DISCUSSION TITLE
How to get records creator as plain text (and not as id)

### DISCUSSION THREAD

**Discussion by fabiante** [pocketbase/pocketbase#541] [Q&A] [answered]
Hi,

I'm playing arround with Pocketbase to build some simple features for my personal website. Using it is a lot of fun 👍🏻 

I want to have a collection where the collection resembles a message list and the messages creator (logged in user) should be displayed in the UI. Is there a simple way to map ressources creator property onto a human readable representation of the users name?

As of now I have not found anything suitable in the API description.

*Edit:* I know of the `/users` endpoints. I of course can think of a solution using these to fetch user data by their user id, but would have hoped that I don't have to do that for every record displayed.

**Comment by ganigeorgiev** [comment, marked as answer]
You can create a "relation" type field in your `messages` collection pointing to the system `profiles` collection (there is a `userId` field that you can use in your collection API rules for access checks if you need that).

By default, the `profiles` collection comes with "name" and "avatar" fields but you can change that by going to "Users > Edit profiles collection (settings cogwheel)".

To fetch all messages with the associated profiles in a single request, you can use the `expand` query parameter:

```js
const result = await client.records.getFullList("messages", 200, { "expand": "yourProfilesRelationFieldName" });
```

Note that in order to be able to view the expanded profiles, **you have to enable and configure the "view" API rule for the `profiles` collection**. For example you can set it to `@request.user.id != ""` if you want to allow only authenticated users to be able to view a user profile (or leave it empty if you want everyone to be able to do it).

---

Please also note, that this will change and will be simplified as part of the ongoing users and profiles refactoring in #376.

**Reply by fabiante** [reply]
Thanks. Tha workes! I will keep an eye on the mentioned refactorings.
